doc/howto/write_a_view/index.md

```diff
@@ -0,0 +1,429 @@
+# How to write a view {#howto_write_a_view}
+
+[TOC]
+
+This HowTo documents how to write a view using the standard library and some helpers from SeqAn3.
+
+\tutorial_head{Difficult, 120 min, \ref tutorial_concepts\,\ref tutorial_ranges, }
+
+\note
+Some of the links from this HowTo only resolve in the developer documentation, because
+they referred to entities from the seqan3::detail namespace.
+We recommend you open this tutorial from http://docs.seqan.de/seqan/3.0.0-master-dev/
+\endnote
+
+# Motivation
+
+We have introduced "views" in \ref tutorial_ranges.
+You can do many things with the views provided by the standard library and those shipped with SeqAn, but in
+certain situations you will want to define your own view.
+This page will teach you the basics of defining your own view.
+
+
+# What makes a view?
+
+A view is a type of `std::ranges::Range` that also models `std::ranges::View`.
+The additional requirements of `std::ranges::View` can be vaguely summarised as "not holding any own data" or at least
+not holding data that is relative in size to the number of elements in the view (e.g. a vector cannot be a view,
+because it's size in memory depends on the number of elements it represents).
+
+A simple view is `std::ranges::subrange`. It can be constructed from a pair of iterators or more precisely an
+*iterator* and a *sentinel*, i.e. the "iterator" representing the end need not be of the same type as the iterator
+representing the beginning – as long as they are comparable.
+This view then holds exactly the iterator-sentinel-pair as its state and nothing else.
+
+But this type does not yet facilitate any "composing-behaviour" that you have seen in \ref tutorial_ranges tutorial,
+`std::ranges::subrange` is simply a type that can be constructed from an iterator-sentinel-pair, you cannot "pipe"
+anything into it.
+Most views are adaptors on other views, e.g. `std::ranges::transform_view` wraps an existing view and applies
+an element-wise transformation on-demand.
+You can directly construct `std::ranges::transform_view` from another view, e.g.
+
+```cpp
```

Comment:
Snippet

Reply:
these snippets don't work yet, because
  1. we don't have transform_view in our std-module
  2. it's not trivial to alias, because like subrange, the deduction won't work

This would make the snippet really terrible to read. I suggest we open a card for finding a better solution to this later.

Reply:
Mhm, well ... I'd kinda expect it to be copy and pasteable. :/